toolchain/base/fixed_size_value_store.h

```diff
@@ -51,6 +53,22 @@ class FixedSizeValueStore {
     return store;
   }
 
+  // Makes a ValueStore of the specified size, initialized to values returned
+  // from a callback. This allows initializing non-copyable values.
+  template <class F>
+    requires requires(F callable) {
+      { callable() } -> std::convertible_to<ValueT>;
+    }
+  static auto MakeWithExplicitSizeFrom(size_t size, F callable)
```

Comment:
Had you considered using `function_ref` here, similar to the constructor below? (also, we typically use `typename` in templates, rather than `class` -- plus `F` is a short typename)

Reply:
From what I've seen in our codebase, we seem to use a template when the function is inline and a function_ref when it's not. I switched it to use std::invoke_result_t in case that helps. And used a better name, thanks.

Reply:
FWIW, I'd suspect there's a different heuristic, but maybe I'm guessing wrong: function_ref when we know the signature that's needed (which should be the case here, right?) and a template when we don't.